### PR TITLE
Add MAGMA option to FSAI

### PR DESCRIPTION
**Pull request by victorapm**
- Allow the use of MAGMA as local linear solver for FSAI.
- Add `HYPRE_FSAISetLocalSolveType` for choosing the local linear solve type used in FSAI.
- Add `HYPRE_BoomerAMGSetFSAILocalSolveType` for the same purpose as above when FSAI is used as a smoother to BoomerAMG.

This PR will be merged into master after #739

### CODE REVIEW

**Review by osborn9 — APPROVED** [review]
Thanks for adding the runtime option @victorapm. Looks good!

File: src/utilities/_hypre_utilities.hpp [hunk 1]
```diff
@@ -602,14 +602,6 @@ using hypre_DeviceItem = sycl::nd_item<3>;
       hypre_assert(0); exit(1);                                                              \
    } } while(0)
 
-#define HYPRE_CUSOLVER_CALL(call) do {                                                       \
```
Comment (rfalgout) [inline src/utilities/_hypre_utilities.hpp:605]: What's this change about @victorapm ?

**Review by rfalgout — APPROVED** [review]
Looks good.  Just the one question.  Thanks!

File: src/utilities/_hypre_utilities.hpp (hunk 1 above)
Comment (victorapm) [reply src/utilities/_hypre_utilities.hpp:605]: We had this defined twice, so I removed the redundant one

Comment (rfalgout) [reply src/utilities/_hypre_utilities.hpp:605]: Got it.  Thanks!

**Pull request merged**